### ISSUE TITLE
Debug logging from ODP miss is too verbose

### ISSUE THREAD

**Issue by bboreham** [weaveworks/weave#1852]
I turned on `--log-level=debug` for the weave router, and got loads of output like this:

````
DEBU: 2016/01/04 14:08:28.916757 ODP miss map[6:BlobFlowKey{type: 6, key: 86dd, mask: ffff} 10:BlobFlowKey{type: 10, key: 14eb14eb, mask: ffffffff} 20:BlobFlowKey{type: 20, key: 00000000, mask: ffffffff} 3:InPortFlowKey{vport: 1} 25:UnknownFlowKey{type: 25, key: 00000000000000000000000000000000, mask: exact} 15:BlobFlowKey{type: 15, key: 00000000, mask: ffffffff} 4:EthernetFlowKey{src: ea:43:f3:4e:e9:07, dst: 33:33:00:01:00:03} 8:BlobFlowKey{type: 8, key: fe80000000000000e843f3fffe4ee907ff020000000000000000000000010003000892bf1100ff00, mask: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff} 19:BlobFlowKey{type: 19, key: 00000000, mask: ffffffff} 2:BlobFlowKey{type: 2, key: 00000000, mask: ffffffff} 23:UnknownFlowKey{type: 23, key: 0000, mask: exact} 24:UnknownFlowKey{type: 24, key: 00000000, mask: exact} 22:UnknownFlowKey{type: 22, key: 00000000, mask: exact}] on port 1
DEBU: 2016/01/04 14:08:28.916884 ODP miss map[22:UnknownFlowKey{type: 22, key: 00000000, mask: exact} 25:UnknownFlowKey{type: 25, key: 00000000000000000000000000000000, mask: exact} 4:EthernetFlowKey{src: 5a:13:72:e5:c3:74, dst: 33:33:00:01:00:03} 8:BlobFlowKey{type: 8, key: fe80000000000000581372fffee5c374ff020000000000000000000000010003000579d41100ff00, mask: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff} 2:BlobFlowKey{type: 2, key: 00000000, mask: ffffffff} 3:InPortFlowKey{vport: 0} 15:BlobFlowKey{type: 15, key: 00000000, mask: ffffffff} 20:BlobFlowKey{type: 20, key: 00000000, mask: ffffffff} 19:BlobFlowKey{type: 19, key: 00000000, mask: ffffffff} 24:UnknownFlowKey{type: 24, key: 00000000, mask: exact} 10:BlobFlowKey{type: 10, key: 14eb14eb, mask: ffffffff} 23:UnknownFlowKey{type: 23, key: 0000, mask: exact} 6:BlobFlowKey{type: 6, key: 86dd, mask: ffff}] on port 0
DEBU: 2016/01/04 14:08:28.917003 ODP miss map[2:BlobFlowKey{type: 2, key: 00000000, mask: ffffffff} 4:EthernetFlowKey{src: 36:c2:79:c6:2f:4a, dst: 33:33:ff:45:f2:57} 19:BlobFlowKey{type: 19, key: 00000000, mask: ffffffff} 22:UnknownFlowKey{type: 22, key: 00000000, mask: exact} 24:UnknownFlowKey{type: 24, key: 00000000, mask: exact} 15:BlobFlowKey{type: 15, key: 00000000, mask: ffffffff} 14:BlobFlowKey{type: 14, key: fe80000000000000c481c5fffe45f25736c279c62f4a000000000000, mask: ffffffffffffffffffffffffffffffffffffffffffffffffffffffff} 6:BlobFlowKey{type: 6, key: 86dd, mask: ffff} 8:BlobFlowKey{type: 8, key: fe8000000000000034c279fffec62f4aff0200000000000000000001ff45f257000000003a00ff00, mask: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff} 25:UnknownFlowKey{type: 25, key: 00000000000000000000000000000000, mask: exact} 12:BlobFlowKey{type: 12, key: 8700, mask: ffff} 3:InPortFlowKey{vport: 1} 23:UnknownFlowKey{type: 23, key: 0000, mask: exact} 20:BlobFlowKey{type: 20, key: 00000000, mask: ffffffff}] on port 1
DEBU: 2016/01/04 14:08:28.929701 ODP miss map[23:UnknownFlowKey{type: 23, key: 0000, mask: exact} 24:UnknownFlowKey{type: 24, key: 00000000, mask: exact} 10:BlobFlowKey{type: 10, key: 00440043, mask: ffffffff} 20:BlobFlowKey{type: 20, key: 00000000, mask: ffffffff} 4:EthernetFlowKey{src: 36:c2:79:c6:2f:4a, dst: ff:ff:ff:ff:ff:ff} 6:BlobFlowKey{type: 6, key: 0800, mask: ffff} 19:BlobFlowKey{type: 19, key: 00000000, mask: ffffffff} 2:BlobFlowKey{type: 2, key: 00000000, mask: ffffffff} 15:BlobFlowKey{type: 15, key: 00000000, mask: ffffffff} 22:UnknownFlowKey{type: 22, key: 00000000, mask: exact} 7:BlobFlowKey{type: 7, key: 00000000ffffffff11c04000, mask: ffffffffffffffffffffffff} 3:InPortFlowKey{vport: 1} 25:UnknownFlowKey{type: 25, key: 00000000000000000000000000000000, mask: exact}] on port 1
````

This rate of logging is likely to make debug level unusable.  

The `--pktdebug` option was created to separate per-packet logging from other information.

**Comment by dpw** [comment]
Note that this is not per-packet logging.  Most misses(*) will result in the creation of flows, limiting the rate of further misses.  A misapprehension that it is per-packet would lead to a exaggerated perception of its volume (and it does not measurably impact the performance of fastdp).

While the output is certainly verbose, I'm not sure that it is surprising that debug logging should be so, and the verbosity has a corresponding value in terms of being able to identify or rule out fastdp as a source of problems.

(*) Though my impression of recent changes suggests this might no longer be true when encrypting.

**Comment by bboreham** [comment]
My impression is that the miss handler runs on every packet when weave is unable to create a fastdp connection.

**Comment by dpw** [comment]
Ah, could be.  If so, the appropriate fix may be to limit the logging in such a case and for the encrypted case, rather than doing something that would lose these messages in the more usual case.

**Comment by brb** [comment]
Closed by #2417
